src/rgw/rgw_lc.cc

```diff
@@ -1556,6 +1556,8 @@ std::string s3_expiration_header(
 
   boost::optional<ceph::real_time> expiration_date;
   boost::optional<std::string> rule_id;
+  /* return the first rule to be deleted */
+  boost::optional<std::string> earliest_rule_id;
```

Comment:
Maybe we can re-use the rule_id. We don't need to update rule_id in 
```
rule_expiration_date =
boost::optional<ceph::real_time>(
ceph::from_iso_8601(rule.get_expiration().get_date()));
rule_id = boost::optional<std::string>(id);
```
We just need to update it when we update the earliest expiration.

Reply:
@zhangsw 
Agree, I changed it